packages/rum/src/resourceUtils.ts

```diff
@@ -90,9 +90,13 @@ export function isValidResource(url: string, configuration: Configuration) {
 
 function isBrowserAgentRequest(url: string, configuration: Configuration) {
   return (
-    startsWith(url, configuration.logsEndpoint) ||
-    startsWith(url, configuration.rumEndpoint) ||
-    startsWith(url, configuration.traceEndpoint) ||
-    (configuration.internalMonitoringEndpoint && startsWith(url, configuration.internalMonitoringEndpoint))
+    haveSameOrigin(url, configuration.logsEndpoint) ||
+    haveSameOrigin(url, configuration.rumEndpoint) ||
+    haveSameOrigin(url, configuration.traceEndpoint) ||
+    (configuration.internalMonitoringEndpoint && haveSameOrigin(url, configuration.internalMonitoringEndpoint))
   )
 }
+
+function haveSameOrigin(url1: string, url2: string) {
+  return startsWith(url2, new URL(url1).origin)
```

Comment:
startsWith("https://rum-intake.com.zogzog.net", new URL('https://rum-intake.com/foo?bar=baz').origin)

Reply:
nice catch, it could be an issue if we reuse that in another context